### PR TITLE
[ADD] open_academy: new module to manage a course

### PR DESCRIPTION
**Pull request by axlongines**
**[ADD] open_academy: new module to manage a course**

*This pr contains the exercise of creation of module, also implements the following views*
- tree view for the main view
- editing, deleting, creating form view

### CODE REVIEW

**Comment by axlongines** [comment]
hi @edy1192, could you review plis :3 ?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (edy1192) [inline .vscode/settings.json]: Remove this file and the .vscode folder please.

File: README.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+*Readme of the tecnical course*
```
Comment (edy1192) [inline README.rst:1]: Please add some description.

File: controllers/controllers.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 # from odoo import http
```
Comment (edy1192) [inline controllers/controllers.py]: Please don't add commented code. Remove this file and all the references to it. When you need this you'll add it.

File: controllers/__init__.py [hunk 4]
```diff
@@ -1,3 +1 @@
-# -*- coding: utf-8 -*-
-
-from . import controllers
\ No newline at end of file
+from . import controllers
```
Comment (edy1192) [inline controllers/__init__.py]: Remove this file also remove the controllers folder. When you need this you'll add it.

File: __init__.py [hunk 5]
```diff
@@ -1,4 +1,2 @@
-# -*- coding: utf-8 -*-
-
 from . import controllers
```
Comment (edy1192) [inline __init__.py]: Remove this import please. Don't need controllers.

File: __manifest__.py [hunk 6]
```diff
@@ -1,35 +1,24 @@
...
     # for the full list
-    'category': 'Uncategorized',
-    'version': '0.1',
-
+    "category": "Uncategorized",
+    "version": "15.0.1.0.0",
     # any module necessary for this one to work correctly
-    'depends': ['base'],
-
+    "depends": ["base"],
```
Comment (edy1192) [inline __manifest__.py]:
```suggestion
    "depends": [
    "base",
    ],
```

File: __manifest__.py [hunk 7]
```diff
@@ -1,35 +1,24 @@
...
-    'data': [
-        # 'security/ir.model.access.csv',
-        'views/views.xml',
-        'views/templates.xml',
+    "data": [
+        'security/ir.model.access.csv',
+        "views/views.xml",
+        "views/templates.xml",
+        "views/course_view.xml",
+        "views/course_menu_view.xml",
```
Comment (edy1192) [inline __manifest__.py]: Please rename the files. See [guideline](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#15file-naming) for file naming please.

File: __manifest__.py [hunk 8]
```diff
@@ -1,35 +1,24 @@
...
     ],
     # only loaded in demonstration mode
-    'demo': [
-        'demo/demo.xml',
-    ],
+    "demo": ["demo/course.xml"],
```
Comment (edy1192) [inline __manifest__.py]:
```suggestion
    "demo": [
    "demo/course.xml"
    ],
```

Please rename the file. See [guideline](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#15file-naming) for file naming please.

File: models/models.py [hunk 9]
```diff
@@ -1,5 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (edy1192) [inline models/models.py:1]: Please don't add commented code. Remove this file and all the references to it.

File: models/__init__.py [hunk 10]
```diff
@@ -1,3 +1,3 @@
-# -*- coding: utf-8 -*-
 
-from . import models
\ No newline at end of file
+from . import models
```
Comment (edy1192) [inline models/__init__.py]: Remove this please.

File: models/course.py [hunk 11]
```diff
@@ -0,0 +1,9 @@
+from odoo import models, fields
```
Comment (edy1192) [inline models/course.py]: Please rename the file. See [guideline](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#15file-naming) for file naming please.

File: models/course.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+
+
+class Course(models.Model):
```
Comment (edy1192) [inline models/course.py]: Class name same as model please.

```suggestion
class OpenAcademyCourse(models.Model):
```

File: security/ir.model.access.csv [hunk 13]
```diff
@@ -1,2 +1,2 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
-access_open_academy_open_academy,open_academy.open_academy,model_open_academy_open_academy,base.group_user,1,1,1,1
\ No newline at end of file
+access_open_academy_model_course,access_open_academy_model_course,model_course,base.group_user,1,1,1,1
```
Comment (edy1192) [inline security/ir.model.access.csv]: Model's name is wrong. Your model is open.academy.course.

```suggestion
access_open_academy_model_course,access_open_academy_model_course,model_open_academy_course,base.group_user,1,1,1,1
```

File: demo/course.xml [hunk 14]
```diff
@@ -0,0 +1,30 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo>
+        <record id="react_course" model="open.academy.ourse">
```
Comment (edy1192) [inline demo/course.xml]:
```suggestion
        <record id="open_academy_course_react_demo" model="open.academy.course">
```

See [guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#254demo-records) to name and xml_id and apply it to all the ids in this file please.

File: demo/demo.xml [hunk 15]
```diff
@@ -27,4 +27,4 @@
           </record>
 -->
     </data>
-</odoo>
\ No newline at end of file
+</odoo>
```
Comment (edy1192) [inline demo/demo.xml]: Do not add empty files. Please remove this.

File: views/course_menu_view.xml [hunk 16]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <menuitem id="list_menu_courses"  name="Courses" sequence="2" action="open_academy_course_action"/>
```
Comment (edy1192) [inline views/course_menu_view.xml]: Plase add a root menu for the module.

```suggestion
    <menuitem id="open_academy_root_menu" name="Open Academy" sequence="1"/>
    <menuitem id="open_academy_course_menu"  name="Courses" sequence="1" action="open_academy_course_action" parent="open_academy_root_menu"/>
```
And see the [guideline](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#252security-view-and-action) to name views and menus

File: views/course_view.xml [hunk 17]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record model="ir.actions.act_window" id="open_academy_course_action">
+        <field name="name">Courses</field>
+        <field name="res_model">course</field>
+        <field name="view_mode">tree,kanban,form</field>
+    </record>
+
+    <record model="ir.ui.view" id="open_academy_course_view_action">
```
Comment (edy1192) [inline views/course_view.xml]: This is not an action. The standard for naming views is <model_name>_view_<view_type> , where view_type is kanban, form, tree, search

```suggestion
    <record model="ir.ui.view" id="open_academy_course_view_tree">
```
See the [guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#252security-view-and-action) please. And apply this to all records in this file please.

File: views/course_view.xml [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="name">List courses</field>
```
Comment (edy1192) [inline views/course_view.xml]: The view name is the same as id but separated by .
```suggestion
        <field name="name">open.academy.course.view.tree</field>
```

File: views/course_view.xml [hunk 19]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record model="ir.actions.act_window" id="open_academy_course_action">
+        <field name="name">Courses</field>
+        <field name="res_model">course</field>
```
Comment (edy1192) [inline views/course_view.xml]: You need to try your changes, this should cause an error when updating the module. Your model name is "open.academy.course"

File: views/course_view.xml [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="view_mode">tree,kanban,form</field>
+    </record>
+
+    <record model="ir.ui.view" id="open_academy_course_view_action">
+        <field name="name">List courses</field>
+        <field name="model">course</field>
```
Comment (edy1192) [inline views/course_view.xml]: You need to try your changes, this should cause an error when updating the module. Your model name is "open.academy.course"

File: views/course_view.xml [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="arch" type="xml">
+            <tree>
+                <field name="title" string="Courses" />
```
Comment (edy1192) [inline views/course_view.xml]: String attibute is not necessary in tree views.

File: views/course_view.xml [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+                <field name="description" string="Description" />
```
Comment (edy1192) [inline views/course_view.xml]: Same here.

File: views/course_view.xml [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="open_academy_course_form_action">
```
Comment (edy1192) [inline views/course_view.xml]: Rename id please.

File: views/course_view.xml [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="name">form courses</field>
```
Comment (edy1192) [inline views/course_view.xml]: Rename form view.

File: views/course_view.xml [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="model">course</field>
```
Comment (edy1192) [inline views/course_view.xml]: Check your model name.

File: views/course_view.xml [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+        <field name="arch" type="xml">
+            <form>
+                <sheet string="Form course">
```
Comment (edy1192) [inline views/course_view.xml]:
```suggestion
                <sheet string="Course">
```

File: views/course_view.xml [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+                    <notebook>
+                        <page name="Course information" string="Course information">
+                            <group col="4">
+                                <field name="title" />
+                                <newline />
+                            </group>
```
Comment (edy1192) [inline views/course_view.xml]: The exercise only asks to add the description field to the page. Please remove the title field from here and add it outside the page.

File: views/templates.xml [hunk 28]
```diff
@@ -21,4 +21,4 @@
         </template>
 -->
     </data>
-</odoo>
```
Comment (edy1192) [inline views/templates.xml:24]: Remove this file please.

File: views/views.xml [hunk 29]
```diff
@@ -57,4 +57,4 @@
               action="open_academy.action_server"/>
 -->
   </data>
-</odoo>
```
Comment (edy1192) [inline views/views.xml:60]: Remove this file please.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by edy1192** [comment]
@AxelRuLo please change the PR's title. 

Use something like this:

[ADD] open_academy: new module to manage an academy

File: views/open_academy_course_views.xml [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+                                <newline />
+                            </group>
+                        </page>
+                    </notebook>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+
```
Comment (edy1192) [inline views/open_academy_course_views.xml]: Just one blank line please.

**Review by edy1192 — CHANGES_REQUESTED** [review]
Please remove the controllers folder, is empty right now.

**Comment by edy1192** [comment]
> @AxelRuLo please change the PR's title.
> 
> Use something like this:
> 
> [ADD] open_academy: new module to manage an academy

This change is pending.

I meant this

image

**Comment by edy1192** [comment]
@AxelRuLo please squash the commits. Just one commit please.

**Comment by edy1192** [comment]
@AxelRuLo please squash the commits. Just one commit please.

**Review by edy1192 — APPROVED** [review]

**Comment by edy1192** [comment]
@axlongines I don't know if something happens that I keep seeing the controllers folder. I'm going to merge the PR and if that folder is still there, please delete it in the next PR.

**Pull request merged**